packages/cosmos-tigergraph/src/index.ts

```diff
@@ -0,0 +1,191 @@
+import { InputNode, InputLink } from '@cosmograph/cosmos'
+
+export class TigerGraphConnection<N extends InputNode, L extends InputLink> {
+  private host: string
+  private graphname: string
+  private username: string
+  private password: string
+  private token: string
+
+  private constructor (host: string, graphname: string, username: string, password: string, token?: string) {
+    this.host = host
+    this.graphname = graphname
+    this.username = username
+    this.password = password
+    this.token = token ?? ''
+  }
+
+  public async generateToken (): Promise<string> {
```

Comment:
@GenericP3rson If you're planning to use the regular Promise syntax (there are no issues with that), we can remove the `async` keywords because they're not necessary.

Reply:
Cool, I'll remove it!